pywikibot/site.py

```diff
@@ -4431,13 +4440,63 @@ def search(self, searchstring, namespaces=None, where="text",
         if not namespaces:
             pywikibot.warning(u"search: namespaces cannot be empty; using [0].")
             namespaces = [0]
-        srgen = self._generator(api.PageGenerator, type_arg="search",
-                                gsrsearch=searchstring, gsrwhat=where,
-                                namespaces=namespaces, step=step,
-                                total=total, g_content=content)
-        if MediaWikiVersion(self.version()) < MediaWikiVersion('1.23'):
-            srgen.request['gsrredirects'] = get_redirects
-        return srgen
+
+        if MediaWikiVersion(self.version()) > MediaWikiVersion('1.11') and checkSnippet:
+
+            if len(namespaces) > 1:
+                raise Error("Cannot do more than one namespace and check snippets")
+
+            # If we want to get the snippets, we have to use a ListGenerator
+            # and manually yield those pages that contain the correct words.
+            srgen = api.ListGenerator("search", site=self,
+                                      srsearch=searchstring, srwhat=where,
+                                      srnamespace=namespaces[0], step=step,
```

Comment:
I believe that `srnamespace` can be an array of namespace numbers.  If that is true, we done need the `Error` above.